core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -472,18 +331,15 @@ public void readFrom(StreamInput in) throws IOException {
         if (in.readBoolean()) {
             scroll = readScroll(in);
         }
-
-        source = in.readBytesReference();
-        extraSource = in.readBytesReference();
+        if (in.readBoolean()) {
+            source = SearchSourceBuilder.PROTOTYPE.readFrom(in);
```

Comment:
add a public static readSearchSourceBuilderFrom to SearchSourceBuilder and make the PROTOTYPE private?